tests/unit/bitcoin-functions.test.ts

```diff
@@ -0,0 +1,103 @@
+import { hexToBytes } from '@noble/hashes/utils';
+import { Transaction, p2tr, p2wpkh } from '@scure/btc-signer';
+import { regtest } from 'bitcoinjs-lib/src/networks';
+
+import {
+  ecdsaPublicKeyToSchnorr,
+  finalizeUserInputs,
+  getInputIndicesByScript,
+} from '../../src/functions/bitcoin/bitcoin-functions';
+import {
+  TEST_ALICE_NATIVE_SEGWIT_PAYMENT_SCRIPT_1,
+  TEST_ALICE_NATIVE_SEGWIT_PUBLIC_KEY_1,
+  TEST_ALICE_NATIVE_SEGWIT_PUBLIC_KEY_2,
+  TEST_ALICE_TAPROOT_PUBLIC_KEY_1,
+  TEST_ALICE_TAPROOT_PUBLIC_KEY_2,
+  TEST_DEPOSIT_PSBT_PARTIALLY_SIGNED_DEPOSIT_PSBT_1,
+  TEST_DEPOSIT_PSBT_PARTIALLY_SIGNED_DEPOSIT_PSBT_2,
+  TEST_WITHDRAW_PSBT_PARTIALLY_SIGNED_WITHDRAW_PSBT_1,
+} from '../mocks/constants';
+
+describe('Bitcoin Functions', () => {
+  describe('getInputIndicesByScript', () => {
+    it('correctly retrieves the input indices by script', () => {
+      const transaction = Transaction.fromPSBT(
+        hexToBytes(TEST_DEPOSIT_PSBT_PARTIALLY_SIGNED_DEPOSIT_PSBT_1)
+      );
+      const aliceScript = hexToBytes(TEST_ALICE_NATIVE_SEGWIT_PAYMENT_SCRIPT_1);
+      const inputIndices = getInputIndicesByScript(aliceScript, transaction);
+      expect(inputIndices).toEqual([0]);
+    });
+
+    it('correctly retrieve an empty array when the script is not found', () => {
+      const transaction = Transaction.fromPSBT(
+        hexToBytes(TEST_WITHDRAW_PSBT_PARTIALLY_SIGNED_WITHDRAW_PSBT_1)
+      );
+      const aliceScript = hexToBytes(TEST_ALICE_NATIVE_SEGWIT_PAYMENT_SCRIPT_1);
+      const inputIndices = getInputIndicesByScript(aliceScript, transaction);
+      expect(inputIndices).toEqual([]);
+    });
```

Comment:
Can there be multiple hits? If yes, add a test for returning multiple. If no, this shouldn't return an array?

Reply:
Yes, there can be multiple, so I'll add a test for that!